test/FunctionalTests/Microsoft.OData.Core.Tests/Json/JsonWriterAsyncBaseTests.cs

```diff
@@ -0,0 +1,243 @@
+        public async Task WritesMixedObjectWithRawValuesCorrectlyAsync()
+        {
+            using (MemoryStream stream = new MemoryStream())
+            {
+                IJsonWriterAsync jsonWriter = CreateJsonWriterAsync(stream, false, Encoding.UTF8);
+                await jsonWriter.StartObjectScopeAsync();
+
+                await jsonWriter.WriteNameAsync("StringProp");
+                await jsonWriter.WriteValueAsync("John");
+
+                await jsonWriter.WriteNameAsync("IntProp");
+                await jsonWriter.WriteValueAsync(10);
+
+                await jsonWriter.WriteNameAsync("BoolPropFalse");
+                await jsonWriter.WriteValueAsync(false);
+
+                await jsonWriter.WriteNameAsync("DateProp");
+                await jsonWriter.WriteValueAsync(new Date(2014, 12, 31));
+
+                await jsonWriter.WriteNameAsync("RawStringProp");
+                await jsonWriter.WriteRawValueAsync(@"""foobar""");
+
+                await jsonWriter.WriteNameAsync("BoolPropTrue");
+                await jsonWriter.WriteValueAsync(false);
+
+                await jsonWriter.WriteNameAsync("RawArrayProp");
+                await jsonWriter.WriteRawValueAsync("[1, 2, 3, 4, 5]");
+
+                await jsonWriter.WriteNameAsync("DateTimeOffsetProp");
+                await jsonWriter.WriteValueAsync(new DateTimeOffset(2014, 12, 31, 12, 42, 30, new TimeSpan(1, 20, 0)));
+
+                await jsonWriter.WriteNameAsync("DateProp");
+                await jsonWriter.WriteValueAsync(new Date(2014, 12, 31));
+
+                await jsonWriter.WriteNameAsync("TimeSpanProp");
+                await jsonWriter.WriteValueAsync(new TimeSpan(12, 42, 30));
+
+                await jsonWriter.WriteNameAsync("TimeOfDayProp");
+                await jsonWriter.WriteValueAsync(new TimeOfDay(12, 42, 30, 100));
+
+                await jsonWriter.WriteNameAsync("ObjectProp");
+                await jsonWriter.StartObjectScopeAsync();
+                await jsonWriter.WriteNameAsync("FloatProp");
+                await jsonWriter.WriteValueAsync(3.1f);
+                await jsonWriter.WriteNameAsync("NestedRawValue");
+                await jsonWriter.WriteRawValueAsync("\"test\"");
+                await jsonWriter.WriteNameAsync("ShortProp");
+                await jsonWriter.WriteValueAsync((short)1124);
+                await jsonWriter.WriteNameAsync("ByteProp");
+                await jsonWriter.WriteValueAsync((byte)10);
+                await jsonWriter.WriteNameAsync("LongProp");
+                await jsonWriter.WriteValueAsync(234234L);
+                await jsonWriter.WriteNameAsync("SignedByteProp");
+                await jsonWriter.WriteValueAsync((sbyte)-10);
+                await jsonWriter.WriteNameAsync("GuidProp");
+                await jsonWriter.WriteValueAsync(new Guid("00000012-0000-0000-0000-012345678900"));
+                await jsonWriter.WriteNameAsync("ArrayPropWithEveryValueRaw");

```

Comment:
`ArrayPropWithEveryValueRaw`? It'd seem to me like the values in the array alternate between non-raw and raw...

Reply:
Hmm. Good point. I think the naming does reflect what was intended.

Reply:
Or did the test not reflect the what the naming intended 🤔

Reply:
I think the name should have been `ArrayProperWithEveryOtherValueRaw`